app/controllers/projects_controller.rb

```diff
@@ -63,20 +63,28 @@ def index
                      current_power.samples
                    end
 
-        @projects = if only_library || exclude_library
-                      @samples.group(:project).count
+        # Retrieve a json of projects associated with samples;
+        # augment with number_of_samples, hosts, tissues.
+        # TODO: Why are we passing host / tissue for each individual sample, with a lot of duplication?
+        #       Can't we tally the hosts and tissues up on the server side, since only summary counts are shown on the page?
+        @projects = if only_updatable
+                      current_power.updatable_projects
                     else
-                      # Make sure you still return projects without any samples. Ex: Project listing
-                      # used by the CLI.
-                      (only_updatable ? current_power.updatable_projects : current_power.projects).map do |p|
-                        [p, Sample.where(project: p).count]
```

Comment:
this was doing a query per project. Also the behavior of including empty projects is probably desired for all cases, not just in the else clause